### ISSUE TITLE
Assertion failed

### ISSUE THREAD

**Issue by domske** [Zylann/godot_heightmap_native_plugin#39]
In Godot 3.0.3 I get the error:
```text
 res://addons/zylann.hterrain/tools/plugin.gd:274 - Assertion failed.
```
Installed from AssertLib. Heightmap Terrain Version: 0.6

**Comment by Zylann** [comment]
What did you do for this to happen? Can you provide the log?

**Comment by domske** [comment]
complete console log:
```text
OpenGL ES 3.0 Renderer: GeForce GTX 1080/PCIe/SSE2
OPENING: C:/Users/User/Documents/Godot/MyTerrain (C:::Users::User::Documents::Godot::MyTerrain)
running cmdline: "C:\Apps\Godot\Godot_v3.0.3-stable_win64.exe" "--path" "C:/Users/User/Documents/Godot/MyTerrain" "--editor"
OpenGL ES 3.0 Renderer: GeForce GTX 1080/PCIe/SSE2
ERROR: _png_error_function: Not a PNG file
   At: drivers/png/image_loader_png.cpp:62
libpng error: Not a PNG file
ERROR: _load_image: PNG Corrupted
   At: drivers/png/image_loader_png.cpp:94
ERROR: _load_mem_png: Condition ' err ' is true. returned: Ref<Image>()
   At: drivers/png/image_loader_png.cpp:252
ERROR: copy_internals_from: Condition ' p_image.is_null() ' is true.
   At: core/image.h:327
ERROR: _jpegd_mem_loader_func: Couldn't initialize ImageLoaderJPG with the given resource.
   At: modules/jpg/image_loader_jpegd.cpp:125
ERROR: _png_error_function: Not a PNG file
   At: drivers/png/image_loader_png.cpp:62
libpng error: Not a PNG file
ERROR: _load_image: PNG Corrupted
   At: drivers/png/image_loader_png.cpp:94
ERROR: _load_mem_png: Condition ' err ' is true. returned: Ref<Image>()
   At: drivers/png/image_loader_png.cpp:252
ERROR: copy_internals_from: Condition ' p_image.is_null() ' is true.
   At: core/image.h:327
ERROR: _jpegd_mem_loader_func: Couldn't initialize ImageLoaderJPG with the given resource.
   At: modules/jpg/image_loader_jpegd.cpp:125
ERROR: _png_error_function: Not a PNG file
   At: drivers/png/image_loader_png.cpp:62
libpng error: Not a PNG file
ERROR: _load_image: PNG Corrupted
   At: drivers/png/image_loader_png.cpp:94
ERROR: _load_mem_png: Condition ' err ' is true. returned: Ref<Image>()
   At: drivers/png/image_loader_png.cpp:252
ERROR: copy_internals_from: Condition ' p_image.is_null() ' is true.
   At: core/image.h:327
ERROR: _jpegd_mem_loader_func: Couldn't initialize ImageLoaderJPG with the given resource.
   At: modules/jpg/image_loader_jpegd.cpp:125
ERROR: _png_error_function: Not a PNG file
   At: drivers/png/image_loader_png.cpp:62
libpng error: Not a PNG file
ERROR: _load_image: PNG Corrupted
   At: drivers/png/image_loader_png.cpp:94
ERROR: _load_mem_png: Condition ' err ' is true. returned: Ref<Image>()
   At: drivers/png/image_loader_png.cpp:252
ERROR: copy_internals_from: Condition ' p_image.is_null() ' is true.
   At: core/image.h:327
ERROR: _jpegd_mem_loader_func: Couldn't initialize ImageLoaderJPG with the given resource.
   At: modules/jpg/image_loader_jpegd.cpp:125
ERROR: _png_error_function: Not a PNG file
   At: drivers/png/image_loader_png.cpp:62
libpng error: Not a PNG file
ERROR: _load_image: PNG Corrupted
   At: drivers/png/image_loader_png.cpp:94
ERROR: _load_mem_png: Condition ' err ' is true. returned: Ref<Image>()
   At: drivers/png/image_loader_png.cpp:252
ERROR: copy_internals_from: Condition ' p_image.is_null() ' is true.
   At: core/image.h:327
ERROR: _jpegd_mem_loader_func: Couldn't initialize ImageLoaderJPG with the given resource.
   At: modules/jpg/image_loader_jpegd.cpp:125
ERROR: _png_error_function: Not a PNG file
   At: drivers/png/image_loader_png.cpp:62
libpng error: Not a PNG file
ERROR: _load_image: PNG Corrupted
   At: drivers/png/image_loader_png.cpp:94
ERROR: _load_mem_png: Condition ' err ' is true. returned: Ref<Image>()
   At: drivers/png/image_loader_png.cpp:252
ERROR: copy_internals_from: Condition ' p_image.is_null() ' is true.
   At: core/image.h:327
ERROR: _jpegd_mem_loader_func: Couldn't initialize ImageLoaderJPG with the given resource.
   At: modules/jpg/image_loader_jpegd.cpp:125
PPATH IS: addons
PPATH IS: addons/zylann.hterrain
PPATH IS: addons/zylann.hterrain
PPATH IS: addons/zylann.hterrain/detail
PPATH IS: addons/zylann.hterrain/detail
PPATH IS: addons/zylann.hterrain
PPATH IS: addons/zylann.hterrain/doc
PPATH IS: addons/zylann.hterrain/doc
PPATH IS: addons/zylann.hterrain
PPATH IS: addons/zylann.hterrain
PPATH IS: addons/zylann.hterrain
PPATH IS: addons/zylann.hterrain
PPATH IS: addons/zylann.hterrain
PPATH IS: addons/zylann.hterrain
PPATH IS: addons/zylann.hterrain
PPATH IS: addons/zylann.hterrain
PPATH IS: addons/zylann.hterrain/shaders
PPATH IS: addons/zylann.hterrain
PPATH IS: addons/zylann.hterrain/src
PPATH IS: addons/zylann.hterrain/src
PPATH IS: addons/zylann.hterrain/src
PPATH IS: addons/zylann.hterrain/src
PPATH IS: addons/zylann.hterrain/src
PPATH IS: addons/zylann.hterrain/src
PPATH IS: addons/zylann.hterrain/src
PPATH IS: addons/zylann.hterrain/src
PPATH IS: addons/zylann.hterrain/src
PPATH IS: addons/zylann.hterrain/src
PPATH IS: addons/zylann.hterrain/src
PPATH IS: addons/zylann.hterrain/src
PPATH IS: addons/zylann.hterrain/src
PPATH IS: addons/zylann.hterrain/src
PPATH IS: addons/zylann.hterrain/src
PPATH IS: addons/zylann.hterrain/src
PPATH IS: addons/zylann.hterrain/src/util
PPATH IS: addons/zylann.hterrain/src/util
PPATH IS: addons/zylann.hterrain/src/util
PPATH IS: addons/zylann.hterrain/src/util
PPATH IS: addons/zylann.hterrain/src/util
PPATH IS: addons/zylann.hterrain/src/util
PPATH IS: addons/zylann.hterrain/src/util
PPATH IS: addons/zylann.hterrain/src/util
PPATH IS: addons/zylann.hterrain
PPATH IS: addons/zylann.hterrain/tools
PPATH IS: addons/zylann.hterrain/tools/brush
PPATH IS: addons/zylann.hterrain/tools/brush
PPATH IS: addons/zylann.hterrain/tools/brush
PPATH IS: addons/zylann.hterrain/tools/brush
PPATH IS: addons/zylann.hterrain/tools
PPATH IS: addons/zylann.hterrain/tools/detail_editor
PPATH IS: addons/zylann.hterrain/tools/detail_editor
PPATH IS: addons/zylann.hterrain/tools/detail_editor
PPATH IS: addons/zylann.hterrain/tools
PPATH IS: addons/zylann.hterrain/tools/generator
PPATH IS: addons/zylann.hterrain/tools/generator
PPATH IS: addons/zylann.hterrain/tools/generator
PPATH IS: addons/zylann.hterrain/tools/generator
PPATH IS: addons/zylann.hterrain/tools
PPATH IS: addons/zylann.hterrain/tools/icons
PPATH IS: addons/zylann.hterrain/tools/icons
PPATH IS: addons/zylann.hterrain/tools/icons
PPATH IS: addons/zylann.hterrain/tools/icons
PPATH IS: addons/zylann.hterrain/tools/icons
PPATH IS: addons/zylann.hterrain/tools/icons
PPATH IS: addons/zylann.hterrain/tools/icons
PPATH IS: addons/zylann.hterrain/tools/icons
PPATH IS: addons/zylann.hterrain/tools/icons
PPATH IS: addons/zylann.hterrain/tools/icons
PPATH IS: addons/zylann.hterrain/tools/icons
PPATH IS: addons/zylann.hterrain/tools/icons
PPATH IS: addons/zylann.hterrain/tools/icons
PPATH IS: addons/zylann.hterrain/tools/icons
PPATH IS: addons/zylann.hterrain/tools/icons
PPATH IS: addons/zylann.hterrain/tools/icons
PPATH IS: addons/zylann.hterrain/tools/icons
PPATH IS: addons/zylann.hterrain/tools/icons
PPATH IS: addons/zylann.hterrain/tools/icons
PPATH IS: addons/zylann.hterrain/tools/icons
PPATH IS: addons/zylann.hterrain/tools/icons
PPATH IS: addons/zylann.hterrain/tools/icons
PPATH IS: addons/zylann.hterrain/tools/icons
PPATH IS: addons/zylann.hterrain/tools/icons
PPATH IS: addons/zylann.hterrain/tools/icons
PPATH IS: addons/zylann.hterrain/tools/icons
PPATH IS: addons/zylann.hterrain/tools
PPATH IS: addons/zylann.hterrain/tools/inspector
PPATH IS: addons/zylann.hterrain/tools/inspector
PPATH IS: addons/zylann.hterrain/tools
PPATH IS: addons/zylann.hterrain/tools
PPATH IS: addons/zylann.hterrain/tools
PPATH IS: addons/zylann.hterrain/tools
PPATH IS: addons/zylann.hterrain/tools
PPATH IS: addons/zylann.hterrain/tools
PPATH IS: addons/zylann.hterrain/tools
PPATH IS: addons/zylann.hterrain/tools
PPATH IS: addons/zylann.hterrain/tools/texture_editor
PPATH IS: addons/zylann.hterrain/tools/texture_editor
PPATH IS: addons/zylann.hterrain/tools/texture_editor
PPATH IS: addons/zylann.hterrain/tools/texture_editor
PPATH IS: addons/zylann.hterrain/tools/texture_editor
PPATH IS: addons/zylann.hterrain/tools/texture_editor
PPATH IS: addons/zylann.hterrain/tools/texture_editor
PPATH IS: addons/zylann.hterrain
PPATH IS: addons/zylann.hterrain/util
PPATH IS: addons/zylann.hterrain/util
PPATH IS: addons/zylann.hterrain/util
PPATH IS: addons/zylann.hterrain/util
PPATH IS: addons/zylann.hterrain/util
erasing C:\Users\User\AppData\Local\Temp/Godot/tmp_asset_231.zip
Heightmap plugin Enter tree
Create HeightMap
Enter world
Enter tree
Edit [Spatial:27197]
Transform changed
SCRIPT ERROR: paint_completed: Assertion failed.
          At: res://addons/zylann.hterrain/tools/plugin.gd:274
```

**Comment by domske** [comment]
I tested it with Godot 3.0.2 too. Same error. Idk. A few days ago I successfully tested the same setup. But I do not remember the terrain version.

**Comment by Zylann** [comment]
It happens because you didn't assign data to the terrain node. Here is how to do it: https://github.com/Zylann/godot_heightmap_native_plugin/blob/master/addons/zylann.hterrain/doc/main.md

Also I think I should disable some callbacks when there is no data so that assertion won't happen.

**Comment by domske** [comment]
Thanks for your fast support. This was the reason. I think I tried a demo a few days ago. I'm a newbie with Godot. First time I heard about ressouce files and the creation from the Inspector. Thanks again. 😄

**Comment by Zylann** [comment]
Should be fixed by 435e09dfc4c0426ff896cfcf2270ef478963892c
